src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ISpeculationAnalyzer.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.Shared.Utilities
+{
+    internal interface ISpeculationAnalyzer
```

Comment:
helper interface as working with AbstractSpeculationAnalyzer directly is a PITA (due to needing to supply 7 type arguments to it).